src/test/java/core/basesyntax/service/FruitTransactionServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.BalanceOperationActivities;
+import core.basesyntax.strategy.OperationActivities;
+import core.basesyntax.strategy.PurchaseOperationActivities;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class FruitTransactionServiceImplTest {
+    private static final Map<FruitTransaction.Operation, OperationActivities>
+            operationStrategyMap = new HashMap<>();
+    private OperationStrategy operationStrategy;
+    private FruitTransactionService fruitTransactionService;
+
+    @BeforeAll
+    static void beforeAll() {
+        operationStrategyMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperationActivities());
+        operationStrategyMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationActivities());
+    }
+
+    @AfterEach
+    void afterEachTest() {
+        Storage.fruitTransactions.clear();
+    }
+
+    @Test
+    void calcFruitTransaction_readData_OK() {
+        Map<String, List<FruitTransaction>> expected = createFruits();
+        operationStrategy = new OperationStrategyImpl(operationStrategyMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(operationStrategy);
+        Map<String, List<FruitTransaction>> result =
+                fruitTransactionService.readFruitTransaction();
+        boolean isEqualsMap = areEqualsMapFruits(expected, result);
+        Assert.assertTrue("Error generating map from storage!", isEqualsMap);
+    }
+
+    @Test
+    void calcFruitTransaction_calcData_OK() {
+        Map<String, Integer> expected = createMapResult();
+        operationStrategy = new OperationStrategyImpl(operationStrategyMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(operationStrategy);
+        Map<String, List<FruitTransaction>> transactionMap = createFruits();
+        Map<String, Integer> result =
+                fruitTransactionService.calcFruitTransaction(transactionMap);
+        boolean isEqualsMap = areEqualsMapFruits(expected, result);
+        Assert.assertTrue("Transaction map calculation error!", isEqualsMap);
+    }
+
+    @Test
+    void calcFruitTransaction_calcData_NotOK() {
+        operationStrategy = new OperationStrategyImpl(operationStrategyMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(operationStrategy);
+        Map<String, List<FruitTransaction>> transactionMap = createFruits_NotOK();
+        Throwable thrown = Assert.assertThrows(RuntimeException.class, () -> {
+            fruitTransactionService.calcFruitTransaction(transactionMap);
+        });
+        Assert.assertNotNull(thrown.getMessage());
+    }
+
+    public Map<String, List<FruitTransaction>> createFruits() {
```

Comment:
**For all methods:** if you are not going to call it outside the class - make it private
```suggestion
    private Map<String, List<FruitTransaction>> createFruits() {
```